spring-auto-restdocs-core/src/test/java/capital/scalable/restdocs/response/ResponseModifyingPreprocessorsTest.java

```diff
@@ -0,0 +1,107 @@
+import static java.util.Collections.singletonList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.MediaType.IMAGE_JPEG_VALUE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.restdocs.operation.OperationResponse;
+import org.springframework.restdocs.operation.preprocess.OperationPreprocessor;
+import org.springframework.restdocs.operation.preprocess.OperationResponsePreprocessor;
+
+public class ResponseModifyingPreprocessorsTest {
+
+    @Test
+    public void shouldShortenJsonArraysWithCombinedShortener() {
+        // given
+        ObjectMapper mapper = new ObjectMapper();
+        OperationResponse unprocessedResponse =
+                new OperationResponseFake("[1,2,3,4]", APPLICATION_JSON_VALUE);
+        // when
+        OperationResponsePreprocessor preprocessor =
+                ResponseModifyingPreprocessors.shortenContent(mapper);
+        OperationResponse processedResponse = preprocessor.preprocess(unprocessedResponse);
+        // then
+        assertThat(processedResponse.getContentAsString(), is("[1,2,3]"));
+    }
+
+    @Test
+    public void shouldShortenBinaryContentWithCombinedShortener() {
+        // given
+        ObjectMapper mapper = new ObjectMapper();
+        OperationResponse unprocessedResponse =
+                new OperationResponseFake("does not matter", IMAGE_JPEG_VALUE);
+        // when
+        OperationResponsePreprocessor preprocessor =
+                ResponseModifyingPreprocessors.shortenContent(mapper);
+        OperationResponse processedResponse = preprocessor.preprocess(unprocessedResponse);
+        // then
+        assertThat(processedResponse.getContentAsString(), is("<binary>"));
+    }
+
+    @Test
+    public void shouldShortenJsonArrays() {
+        // given
+        ObjectMapper mapper = new ObjectMapper();
+        OperationResponse unprocessedResponse =
+                new OperationResponseFake("[1,2,3,4]", APPLICATION_JSON_VALUE);
+        // when
+        OperationPreprocessor preprocessor =
+                ResponseModifyingPreprocessors.limitJsonArrayLength(mapper);
+        OperationResponse processedResponse = preprocessor.preprocess(unprocessedResponse);
+        // then
+        assertThat(processedResponse.getContentAsString(), is("[1,2,3]"));
+    }
+
+    @Test
+    public void shouldShortenBinary() {
+        // given
+        OperationResponse unprocessedResponse =
+                new OperationResponseFake("does not matter", IMAGE_JPEG_VALUE);
+        // when
+        OperationPreprocessor preprocessor =
+                ResponseModifyingPreprocessors.replaceBinaryContent();
+        OperationResponse processedResponse = preprocessor.preprocess(unprocessedResponse);
```

Comment:
you could extract these tests to shared method and just change input params: preprocessor, input, expected output